src/tenant/actions.py

```diff
@@ -0,0 +1,143 @@
+    # Say friend and enter (Doors of Durin)
+    keyword = forms.CharField(widget=forms.HiddenInput)
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        confirmation = cleaned_data.get("confirmation", "")
+        keyword = cleaned_data.get("keyword", "")
+        if confirmation and confirmation != keyword:
+            self.add_error("confirmation", _("The confirmation does not match the keyword"))
+
+        return cleaned_data
+
+
+@action(
+    permissions=["delete"],
+    description="Delete selected %(verbose_name_plural)s",
+)
+def delete_selected(modeladmin, request, queryset):
+    """
+    Default action which deletes the selected objects.
+
+    This action first displays a confirmation page which shows all the
+    deletable objects, or, if the user has no permission one of the related
+    childs (foreignkeys), a "permission denied" message.
+
+    Next, it deletes all selected objects and redirects back to the change list.
+    """
+    opts = modeladmin.model._meta
+    app_label = opts.app_label
+
+    objects = modeladmin.model.objects.filter(
+        pk__in=[str(x) for x in request.POST.getlist(helpers.ACTION_CHECKBOX_NAME)]).exclude(
+            schema_name=get_public_schema_name())
+
+    # Populate deletable_objects, a data structure of all related objects that
+    # will also be deleted.
+    (
+        deletable_objects,
+        model_count,
+        perms_needed,
+        protected,
+    ) = modeladmin.get_deleted_objects(objects, request)
+
+    # Do the deletion and return None to display the change list view again.
+    if request.POST.get("post") and not protected:  # The user has confirmed the deletion.
+        if perms_needed:
+            raise PermissionDenied
+        form = DeleteSelectedConfirmationForm(data=request.POST)
+        if form.is_valid():
+            n = len(queryset)
+            if n:
+                for obj in queryset:
+                    obj_display = str(obj)
+                    modeladmin.log_deletion(request, obj, obj_display)
+                modeladmin.delete_queryset(request, queryset)
+                modeladmin.message_user(
+                    request,
+                    "Successfully deleted %(count)d %(items)s."
+                    % {"count": n, "items": model_ngettext(modeladmin.opts, n)},
+                    messages.SUCCESS,
+                )
+
+            # return None to display the change list page again.
+            return None
+    else:
+        # create form and pass the data which objects were selected before triggering 'post' action.
+        # '_selected_action' (ACTION_CHECKBOX_NAME) is required for the admin intermediate form to submit
+        form = DeleteSelectedConfirmationForm(
+            initial={
+                helpers.ACTION_CHECKBOX_NAME: objects.values_list("id", flat=True),
+                # generate "sekret" keyword from two random words
+                "keyword": "/".join(generate(2)),
```

Comment:
Instead of adding a new dependancy, can we just type the deck name for confirmation?  Similar to when you delete a github repo?  This is what is requested in issue #1255.

Reply:
there multiple objects (hencse it admin actions), but which one I should use as keyword / passphrase? first one, or just random object from queryset?

Reply:
@Bashu, can we only allow a single deck at a time?  It can be an admin action (one selection only) or an action on the detail view of the item.